src/app/my-account/my-account.component.ts

```diff
@@ -0,0 +1,105 @@
+    this.passFields = { oldPassField: '', newPassField: '', verifyPassField: '' };
+   }
+
+  public ngOnInit() {
+  }
+
+  public oldPassChange(): void {
+    this.passColors.oldPass = this.colors.blue;
+  }
+  public newPassChange(): void {
+    this.passColors.newPass = this.passwordStrength();
+    this.verifyPassChange();
+  }
+  public verifyPassChange(): void {
+    if (this.passFields.verifyPassField === '' && this.passFields.newPassField === '') { this.passColors.verifyPass = this.colors.blue; }
+    else if (this.passFields.verifyPassField === this.passFields.newPassField) { this.passColors.verifyPass = this.colors.green; }
+    else { this.passColors.verifyPass = this.colors.red; }
+  }
+  public onSubmit(): void {
+    this.submitDisabled = true;
+    this.messages.display = true;
+    let warn: boolean = true;
+    if (this.passFields.oldPassField === '') {
+      this.passColors.oldPass = this.colors.red;
+      this.messages.message = 'You Must Enter Your Old Password/Passphrase!';
+    } else if (this.passFields.newPassField === '') {
+      this.messages.message = 'You Must Enter A New Password/Passphrase!';
+    } else if (this.passColors.newPass === this.colors.red) {
+      this.messages.message = 'Choose A Stronger Password/Passphrase!';
+    } else if (this.passFields.verifyPassField !== this.passFields.newPassField) {
+      this.messages.message = 'New Passwords/Passphrases Do Not Match!';
+    } else if (this.passFields.oldPassField === this.passFields.newPassField) {
+      this.messages.message = 'Choose A New Password/Passphrase!';
+    } else if (!this.oldPassMatches()) {
+      this.passColors.oldPass = this.colors.red;
+      this.messages.message = 'Old Password/Passphrase Is Invalid!';
+    } else {
+      // TODO: REMOVE the following line when we connect to DB
+      this.pword = this.passFields.newPassField;
+      this.messages.message = 'Password/Passphrase Changed';
+      warn = false;
+      this.passColors.oldPass = this.colors.green;
+    }
+    this.animWarn(warn);
+    this.submitDisabled = false;
+    console.log(this.pword);
+  }
+
+  // TODO: CHANGE this to check with actual password in DB
+  private oldPassMatches(): boolean {
+    return this.passFields.oldPassField === this.pword;
+  }
+  private passwordStrength(): number {
+    if (this.passFields.newPassField === '') { return this.colors.blue; }
+    let total: number = 0;
+    total += this.passFields.newPassField.toUpperCase() !== this.passFields.newPassField ? 1 : 0; // has lower
+    total += this.passFields.newPassField.toLowerCase() !== this.passFields.newPassField ? 1 : 0; // has upper
+    total += /(!|@|#|\$|%|\^|\&|\*|\(|\)|-|\+)/.test(this.passFields.newPassField) ? 1 : 0; // has special
+    total += /\d/.test(this.passFields.newPassField) ? 1 : 0; // has digit
+    total += this.passFields.newPassField.length >= 15 ? 7 : this.passFields.newPassField.length >= 8 ? 5 : 0; // length at least 20
```

Comment:
Does this fulfill VA password requirements? I suspect we must require passphrases to also have multiple character classes.

For passphrases, we want to make sure that it in fact contains letters, not just numbers.

Also, bcrypt ignores characters after the first 72, so we should have a maximum password length of 72.

Reply:
Unfortunately, I haven't heard back about the exact password requirements yet.  So this may need to change later in another ticket, regardless of what it does now.

Reply:
Changed it so passwords must be at least 8 characters (line 80, but a max of 72 (lines 51-52), they must also include 3 of the character classes for yellow or all 4 for green.

Reply:
Found this:
https://www.communityviewer.va.gov/WebHelp/Password_Requirements.htm
So I also added the requirement that there must be at least 6 unique characters.